src/main/java/com/typee/ui/TabPanel.java

```diff
@@ -0,0 +1,40 @@
+package com.typee.ui;
+
+import java.util.logging.Logger;
+
+import com.typee.commons.core.LogsCenter;
+import com.typee.model.Tab;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+
+/**
+ * Panel containing the list of tabs/menus.
+ */
+public class TabPanel extends UiPart<Region> {
+    private static final Logger logger = LogsCenter.getLogger(TabPanel.class);
+    private static final String FXML = "TabPanel.fxml";
+
+    @FXML
+    private ListView<Tab> tabListView;
+
+    public TabPanel(ObservableList<Tab> tabList) {
+        super(FXML);
+        logger.info("Tabs generated: " + tabList.toString());
+        tabListView.setItems(tabList);
+        tabListView.setCellFactory(param -> new ListCell<Tab>() {
+            @Override
+            protected void updateItem(Tab tab, boolean empty) {
+                super.updateItem(tab, empty);
+                if (empty || tab == null || tab.getName() == null) {
+                    setText(null);
+                } else {
+                    setText(tab.getName());
+                }
```

Comment:
Looks good! Maybe rename boolean empty to isEmpty to emphasize that it's a boolean :) You can consider having another method isEmpty that checks the condition in line 32 to adhere to the SLAP principle

Reply:
Thanks for reminding me the SLAP again. Will update the parameter name to isEmpty instead. For the isEmpty() method, the super method force to set it empty and update the list cell object so I don't think it is necessary. Anyway, thanks for the review.

from Cell.java
`   * @param item The new item for the cell.
     * @param empty whether or not this cell represents data from the list. If it
     *        is empty, then it does not represent any domain data, but is a cell
     *        being used to render an "empty" row.
     */
    protected void updateItem(T item, boolean empty) {
        setItem(item);
        setEmpty(empty);
        if (empty && isSelected()) {
            updateSelected(false);
        }
    }`